src/access/acp_v2.ts

```diff
@@ -867,7 +867,7 @@ function copyPolicyExcludingActor(
   const newIriSuffix =
     "_copy_without" +
     `_${encodeURIComponent(actorRelationToExclude)}_${actorToExclude}` +
-    `_${Date.now()}_${Math.random()}`;
+    `_${uuidv4()}`;
```

Comment:
I think all these identifiers for `newIriSuffix` can be removed, and we can just always generate a `uuidv4()` in each of `createResourceMatcherFor` and `createResourcePolicyFor` probably should just be the matcher IRI with the hash replaced with a uuid, rather than appending to the IRI.

e.g., the current implementation may be leaking information about the actor & the relationship in the policy or matcher IRI, when that value carries no significance, and it's really just a "makes the policy nice to read/look at"

Though that may be a wider refactor, so deferring to @NSeydoux

Reply:
So if I understand this comment better, we can replace lines 867-870 with 

`const newIriSuffix = uuidv4();`

Is that correct? 

Will wait for @NSeydoux response before pushing updates. 

Beyond just changing the value to the code mentioned above, would we need to change anything else? Specifically, do we have code in other places that relies on having the "copy_without", "actorRelationToExclude" or "actorTo Exclude" in the IriSuffix?

Reply:
> Beyond just changing the value to the code mentioned above, would we need to change anything else? Specifically, do we have code in other places that relies on having the "copy_without", "actorRelationToExclude" or "actorTo Exclude" in the IriSuffix?

as far as I can tell, no, and in fact, ESS will make those IRIs for subject/object opaque, so they won't be returned back as-is anyway

Reply:
There shouldn't be any hard-coded dependencies on the resource names, or reliance on any in-IRI semantics in the low-level API (we know there is some in the high-level API, but that's irrelevant here). I don't think ESS changes these IRIs, for instance when using the permissions app you can give a human-friendly name to a policy and it is persisted as-is. Finally, I think there isn't any information leak here: any user authorized to see the ACR in the first place would be able to see any actor mentioned in it, so it being embedded in an IRI doesn't add any information to what the resource already contains. It is really just for debug purpose.

All that said, it is a good practice to not have semantics in IRIs, so provided there isn't any reliance on this piece of logic, I think we can go forward with Emelia's suggestion and just use the `uuid`.

Reply:
I've also subsequently removed the rather brittle test for encoding hashes in the ACP IRIs, as we don't do that any more, and the only place this may come in is through existing IRIs maybe.